src/components/Card.js

```diff
@@ -1,61 +1,135 @@
+    } else {
+      cardLikeButton.classList.remove("card__like-button_active");
+    }
+  }
+
+  _fillCardTemplate() {
+    const imageElement = this._cardElement.querySelector(".card__image");
+    imageElement.src = this._link;
+    imageElement.alt = this._name;
+    this._cardElement.id = this._id;
+
+    this._cardElement.querySelector(".card__title").textContent = this._name;
+    this._cardLikesCountElement =
+      this._cardElement.querySelector(".card__likes-count");
+
+    if (this._ownerId !== this._myId) {
+      this._cardElement.querySelector(".card__delete-button").remove();
+    }
+  }
+
+  _setEventListeners() {
+    this._cardElement
+      .querySelector(".card__like-button")
+      .addEventListener("click", () => {
+        this._handleLikeButton(this._id, this.isLiked());
+      });
+    if (this._myId === this._ownerId) {
+      this._cardElement
+        .querySelector(".card__delete-button")
+        .addEventListener("click", () => {
+          this._handleDeleteButtonClick();
+        });
+    }
+    this._cardElement
+      .querySelector(".card__image")
+      .addEventListener("click", () => {
+        this._handlePreviewImage({ name: this._name, link: this._link });
+      });
+  }
+
+  deleteCard() {
+    this._handleDeleteButton(this._id);
   }
 
-  //handleLikeButton
   _toggleLikeButton() {
     this._cardElement
       .querySelector(".card__like-button")
       .classList.toggle("card__like-button_active");
   }
 
-  //handleDeleteButton
-  _deleteCard() {
-    this._cardElement.remove();
-    this._cardElement = null;
+  _hideDeleteButton() {
+    this._cardElement.querySelector(".card__delete-button");
+    if (this._ownerId !== this._myId) {
+      this._cardElement.querySelector(".card__delete-button").remove();
+    }
   }
 
-  _setEventListeners() {
-    const likeButton = this._cardElement.querySelector(".card__like-button");
-    likeButton.addEventListener("click", () => {
-      this._toggleLikeButton();
-    });
-
-    const deleteButton = this._cardElement.querySelector(
-      ".card__delete-button"
-    );
-    deleteButton.addEventListener("click", () => {
-      this._deleteCard();
-    });
+  _getTemplate() {
+    const cardElement = document
+      .querySelector(this._cardSelector)
+      .content.querySelector(".card")
+      .cloneNode(true);
 
-    const cardImage = this._cardElement.querySelector(".card__image");
-    cardImage.addEventListener("click", () => {
-      this._handleCardClick({ name: this._name, link: this._link }); // destructuring the object
-    });
+    return cardElement;
   }
 
   getView() {
     this._cardElement = this._getTemplate();
     this._setEventListeners();
-
-    this._cardImage = this._cardElement.querySelector(".card__image");
-    this._cardImage.src = this._link;
-    this._cardImage.alt = this._name;
+    this._cardElement.querySelector(".card__image").src = this._link;
+    this._cardElement.querySelector(".card__image").alt = this._name;
```

Comment:
**Could be improved**

Please, find it only once in line 121



All elements that are found every time you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `generateCard`) only **1 time** not to waste resources on searching them again and again.

`Class fields` (the so called **class variables** with `this`) are accessible inside any method of the class. You shouldn’t pass **class fields** into the call of any method because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form` and so on.

Reply:
Changed.